tests/test_cli.py

```diff
@@ -136,34 +136,45 @@ def test_cli_custom_dimensions():
     assert result_valid.exit_code == 0
 
 
-def test_cli_custom_dimensions_empty():
-    """Check that CLI raises an error when specifying an empty directory ('empty')"""
+def test_cli_custom_dimensions_fails():
+    """Check that CLI raises an error when specifying a non-existing
+    directory ('foo')"""
 
     result_valid = runner.invoke(
         cli,
         [
             "validate-project",
-            str(TEST_DATA_DIR / "non-default_dimensions_failing"),
+            str(TEST_DATA_DIR / "non-default_dimensions_passing"),
             "--dimensions",
-            "['variable', 'region', 'empty']",
+            "['variable', 'region', 'foo']",
         ],
     )
     assert result_valid.exit_code == 1
 
 
-def test_cli_custom_dimensions_fails():
-    """Check that CLI raises an error when specifying a non-existent
-    directory ('foo')"""
+def test_cli_empty_dimensions_run():
+    """Check that CLI runs through when an empty directory is not specified in
+    custom dimensions"""
 
     result_valid = runner.invoke(
         cli,
         [
             "validate-project",
-            str(TEST_DATA_DIR / "non-default_dimensions_passing"),
+            str(TEST_DATA_DIR / "non-default_dimensions_failing"),
             "--dimensions",
-            "['variable', 'region', 'foo']",
+            "['variable', 'region']",
         ],
     )
+    assert result_valid.exit_code == 0
+
+
+def test_cli_empty_dimensions_fails():
```

Comment:
Since this is a newly created test for this PR, it would be great to not only test for the exit code but for the type of error and error message as well. You could take some inspiration from here: https://github.com/IAMconsortium/nomenclature/blob/main/tests/test_cli.py#L200